code/modules/admin/verbs/mapping.dm

```diff
@@ -67,6 +68,26 @@ GLOBAL_PROTECT(admin_verbs_debug_mapping)
 /obj/effect/debugging/marker/Move()
 	return 0
 
+/client/proc/export_map()
+	set category = "Mapping"
+	set name = "Export Map"
+
+	var/z_level = input("Export Which Z-Level?", "Map Exporter", 2) as num
+	var/start_x = input("Start X?", "Map Exporter", 1) as num
+	var/start_y = input("Start Y?", "Map Exporter", 1) as num
+	var/end_x = input("End X?", "Map Exporter", world.maxx-1) as num
+	var/end_y = input("End Y?", "Map Exporter", world.maxy-1) as num
+	var/date = time2text(world.timeofday, "YYYY-MM-DD_hh-mm-ss")
+	var/file_name = input("Filename?", "Map Exporter", "exportedmap_[date]") as text
+	var/confirm = alert("Are you sure you want to do this? This will cause extreme lag!", "Map Exporter", "Yes", "No")
+
+	if(confirm != "Yes")
+		return
+
+	var map_text = write_map(start_x, start_y, z_level, end_x, end_y, z_level, 24)
+	text2file(map_text, "data/[file_name].dmm")
+	usr << ftp("data/[file_name].dmm", "[file_name].dmm")

```

Comment:
Seems like a possible security issue

@crossedfall would this not allow people to write map files wherever they wanted on the system, as well as fill up the system disk space by continuously writing map files

Reply:
I mean only admins can use this, and we already have access to much worse debug tools. From what I can tell, it writes it then sends it to me,

Reply:
Checked just now, it **does write it to data.** You got that correct.

Reply:
Yeah, not a fan. I'd recommend adding a cooldown to the command and then you could use `fdel` to remove the file after it has been sent to the user.

Also move the file into a special directory in the data folder. (e.g. `data/mapexport/[filename].dmm`)

Reply:
Got it! Fixing now....